### PR TITLE
[SPARK-34313][SQL] Migrate ALTER TABLE SET/UNSET TBLPROPERTIES commands to use UnresolvedTable to resolve the identifier

### PR DESCRIPTION
**Pull request by imback82**
### What changes were proposed in this pull request?

This PR proposes to migrate `ALTER TABLE ... SET/UNSET TBLPROPERTIES` to use `UnresolvedTable` to resolve the table identifier. This allows consistent resolution rules (temp view first, etc.) to be applied for both v1/v2 commands. More info about the consistent resolution rule proposal can be found in [JIRA](https://issues.apache.org/jira/browse/SPARK-29900) or [proposal doc](https://docs.google.com/document/d/1hvLjGA8y_W_hhilpngXVub1Ebv8RsMap986nENCFnrg/edit?usp=sharing).

### Why are the changes needed?

This is a part of effort to make the relation lookup behavior consistent: [SPARK-29900](https://issues.apache.org/jira/browse/SPARK-29900).

### Does this PR introduce _any_ user-facing change?

After this PR, `ALTER TABLE SET/UNSET TBLPROPERTIES` will have a consistent resolution behavior.

### How was this patch tested?

Updated existing tests / added new tests.

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala [hunk 1]
```diff
@@ -431,6 +431,17 @@ class DataSourceV2Strategy(session: SparkSession) extends Strategy with Predicat
       val changes = Seq(TableChange.setProperty(TableCatalog.PROP_LOCATION, location))
       AlterTableExec(table.catalog, table.identifier, changes) :: Nil
 
+    case AlterTableSetProperties(table: ResolvedTable, props) =>
+      val changes = props.map { case (key, value) =>
+        TableChange.setProperty(key, value)
+      }.toSeq
+      AlterTableExec(table.catalog, table.identifier, changes) :: Nil
+
+    // TODO: v2 `UNSET TBLPROPERTIES` should respect the ifExists flag.
```
Comment (imback82) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala:440]: I will handle this after this PR is merged; much easier to handle with new framework.

**Comment by imback82** [comment]
cc @cloud-fan @MaxGekk

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala [hunk 2]
```diff
@@ -3493,14 +3493,16 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with SQLConfHelper with Logg
           relationTypeMismatchHint = Some("Please use ALTER TABLE instead.")),
         cleanedTableProperties)
     } else {
-      AlterTableSetPropertiesStatement(
-        visitMultipartIdentifier(ctx.multipartIdentifier),
+      AlterTableSetProperties(
+        createUnresolvedTable(
+          ctx.multipartIdentifier,
+          "ALTER TABLE ... SET TBLPROPERTIES"),
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala:3499]: let's rebase this PR. `createUnresolvedTable`  takes a new parameter now.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/SQLViewSuite.scala [hunk 3]
```diff
@@ -167,6 +167,12 @@ abstract class SQLViewSuite extends QueryTest with SQLTestUtils {
       assertAnalysisError(
         s"ALTER TABLE $viewName DROP PARTITION (a='4', b='8')",
         s"$viewName is a temp view. 'ALTER TABLE ... DROP PARTITION ...' expects a table")
+      assertAnalysisError(
+        s"ALTER TABLE $viewName SET TBLPROPERTIES ('p' = 'an')",
+        s"$viewName is a temp view. 'ALTER TABLE ... SET TBLPROPERTIES' expects a table")
+      assertAnalysisError(
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/SQLViewSuite.scala:173]: cc @MaxGekk we should add tests here.

**Review by cloud-fan — APPROVED** [review]

File: sql/core/src/test/scala/org/apache/spark/sql/execution/SQLViewSuite.scala (hunk 3 above)
Comment (MaxGekk) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/SQLViewSuite.scala:173]: And I guess to `HiveDDLSuite` too, right? I just wonder if we test v2 and v1 Hive external catalog, should we test v1 In-Memory catalog in that case?

Comment (cloud-fan) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/SQLViewSuite.scala:173]: I don't think so. It's UT and we know how the code works. The catalog implementation doesn't matter. Testing it once with whatever catalog is good enough.

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**